### PR TITLE
Make codeowners more specific

### LINKED ISSUE #850: PR review requests are not following the CODEOWNERS assignements
## Description

Codeowners previous to the monorepo migration are not being followed. I suspect it's due to the last line overrides the more detailed previous ones.

https://github.com/WordPress/openverse/blob/1b79338701344f93a48657a0ee744a03c623e1fe/.github/CODEOWNERS#L1-L4

## Reproduction

See for example: https://github.com/WordPress/openverse/pull/450#issuecomment-1440096233

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #850 by @krysal

## Description

Makes the CODEOWNERS file more specific on which part of the code each subteam has to review, avoiding a wildcard overriding previous assignments.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,37 @@
...
+rfcs/                       @WordPress/openverse-maintainers
+templates/                  @WordPress/openverse-maintainers
+.gitattributes              @WordPress/openverse-maintainers
+.gitignore                  @WordPress/openverse-maintainers
+.pre-commit-config.yaml     @WordPress/openverse-maintainers
+CODE_OF_CONDUCT.md          @WordPress/openverse-maintainers
+CONTRIBUTING.md             @WordPress/openverse-maintainers
+justfile                    @WordPress/openverse-maintainers
+LICENSE                     @WordPress/openverse-maintainers
+README.md
```
Comment (zackkrida) [inline .github/CODEOWNERS:37]:
```suggestion
README.md
```

Is this meant to be assigned to `@WordPress/openverse-maintainers` instead of blank?

Comment (krysal) [reply .github/CODEOWNERS:37]: Right! Good catch, thanks.

**Review by dhruvkb — APPROVED** [review]
This is awesome! However, I wonder if instead of listing the common files at the bottom, can a wildcard `*` at the top suffice?

**Comment by krysal** [comment]
> This is awesome! However, I wonder if instead of listing the common files at the bottom, can a wildcard `*` at the top suffice? –@dhruvkb 

I wondered the same thing, but decided on the side that seems safer first. We can check later if it gets too conbersome to maintain this file.

**Pull request merged**